operator/chartdownloader/factory.go

```diff
@@ -10,7 +10,7 @@ import (
 )
 
 type ChartDownloader interface {
-	Download(context.Context, string) (*chart.Chart, error)
+	Download(context.Context, string, string) (*chart.Chart, error)
```

Comment:
I tend to omit param names when there's only one param per type. But since there's now 2 `string` args, I think param names should be added to make it clear that the chart url comes first, then the version. 

```suggestion
	Download(ctx context.Context, chart string, version string) (*chart.Chart, error)
```